src/Tests/DcsBios/DCSBIOSControlLocatorTests.cs

```diff
@@ -0,0 +1,363 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using ClassLibraryCommon;
+using DCS_BIOS.ControlLocator;
+using DCS_BIOS.Json;
+using DCS_BIOS.Serialized;
+using Xunit;
+
+namespace DCSFPTests.DcsBios
+{
+    public class DCSBIOSControlLocatorTests
+    {
+        //private readonly int _a10C = 5; // A-10C Thunderbolt/II
+        //private readonly string _dcsbiosPath = Environment.ExpandEnvironmentVariables(@"%USERPROFILE%\Saved Games\DCS\Scripts\DCS-BIOS\doc\json");
+        //private readonly string _arc21025KhzSelectorId = "ARC210_25KHZ_SEL";
+
+        [Theory]
+        [InlineData(@"%USERPROFILE%\Saved Games\DCS\Scripts\DCS-BIOS\doc\json", 5, "ARC210_25KHZ_SEL")] // A-10C Thunderbolt/II
+        public void TestLocatorGetControlValidJSONPath(string jsonPath, int aircraftId, string dcsbiosControlId)
+        {
+            // !! Invalid path logs to error log but doesn't throw exception to the outside
+            jsonPath = Environment.ExpandEnvironmentVariables(jsonPath);
+            if (!Directory.Exists(jsonPath))
+            {
+                //No need to test as dcs-bios isn't installed
+                return;
+            }
+            Common.ResetEmulationModesFlag();
+
+            DCSAircraft.FillModulesListFromDcsBios(jsonPath, true);
+            var dcsAircraft = DCSAircraft.GetAircraft(aircraftId);
+
+            DCSBIOSControlLocator.DCSAircraft = dcsAircraft;
+            DCSBIOSControlLocator.JSONDirectory = jsonPath;
+
+            var dcsbiosControl = DCSBIOSControlLocator.GetControl(dcsbiosControlId);
+
+            Assert.Equal(dcsbiosControl.Identifier, dcsbiosControlId);
+        }
+
+        [Theory]
+        [InlineData(@"%USERPROFILE%\Saved Games\DCS\Scripts\DCS-BIOS\doc\json", 5, "NO_CONTROL_WITH_THIS_ID")] // A-10C Thunderbolt/II
+        public void TestLocatorGetControlInvalidControlId(string jsonPath, int aircraftId, string dcsbiosControlId)
+        {
+            jsonPath = Environment.ExpandEnvironmentVariables(jsonPath);
```

Comment:
The "Arrange-Act-Assert" pattern is respected but maybe here, the arrange part (the setup) could be placed in a private function. Between some tests there is a lot of repetive code. (jsonPath, directory exists , fill module list, DCSBIOSControlLocator.DCSAircraft = dcsAircraft..)

Reply:
Done. Much less code. I noticed there was a clash between ```ClassLibraryCommon``` tests and ```DCSBIOSControlLocator``` tests since both sets and reads the ```EmulationModeFlag```.
I added ```[Collection("Sequential")] // This uses EmulationMode``` to both so they run sequentially.